### PR TITLE
Support to upload zypper repository profile

### PR DESCRIPTION
**Pull request by sbernhard**

### CODE REVIEW

**Comment by cnsnyder** [comment]
@candlepin-bot ok to test

File: src/rhsm/profile.py [hunk 1]
```diff
@@ -36,8 +36,16 @@
 except ImportError:
     yum = None
 
+try:
+    import zypp_plugin
```
Comment (jirihnidek) [inline src/rhsm/profile.py:40]: This module is imported but it is not used at all. Style checker complains about it:
```
running flake8
src/rhsm/profile.py:40:5: F401 'zypp_plugin' imported but unused
```
You can solve it using something from this module or you can add following comment to the code:
```python
# noqa: F401
```

Comment (sbernhard) [reply src/rhsm/profile.py]: Thanks!

File: src/rhsm/profile.py [hunk 2]
```diff
@@ -36,8 +36,16 @@
 except ImportError:
     yum = None
 
+try:
+    import zypp_plugin # noqa: F401
```
Comment (jirihnidek) [inline src/rhsm/profile.py:40]: It's better but still not perfect ;-)
```
src/rhsm/profile.py:40:23: E261 at least two spaces before inline comment
```
You can test style of your file using: `flake8 src/rhsm/profile.py` before pushing to server.

Comment (sbernhard) [reply src/rhsm/profile.py]: ah. sorry for that. Thanks for you help. Fixed it now.

**Comment by jirihnidek** [comment]
@candlepin-bot ok to test

**Comment by jirihnidek** [comment]
I have concerns about results of tests on opensuse and sles12. Maybe tests are failing in master too, but we had several PRs yesterday without such failures. These two tests are failing and it is related zypper plugin rhsm:

```
Successfully attached a subscription for: Red Hat Enterprise Linux Atomic Host
Refreshing service 'rhsm'.
Repositories have been successfully exported to /tmp/tmp7sAd06.repo.
F
======================================================================
ERROR: test_can_download_rpm (test_serviceplugin.TestServicePlugin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/jenkins/workspace/subscription-manager/subscription-manager-suse-opensuse42-tests/test/zypper_test/test_serviceplugin.py", line 59, in test_can_download_rpm
    subprocess.check_call('test "$(find /var/cache/zypp/packages/ -name \'{RHSM_TEST_PACKAGE}*.rpm\' | wc -l)" -gt 0'.format(**os.environ), shell=True)
  File "/usr/lib64/python2.7/subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
CalledProcessError: Command 'test "$(find /var/cache/zypp/packages/ -name 'sos*.rpm' | wc -l)" -gt 0' returned non-zero exit status 1

======================================================================
FAIL: test_provides_subman_repos_if_registered_and_subscribed (test_serviceplugin.TestServicePlugin)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/jenkins/workspace/subscription-manager/subscription-manager-suse-opensuse42-tests/test/zypper_test/test_serviceplugin.py", line 46, in test_provides_subman_repos_if_registered_and_subscribed
    self.assertTrue(self.has_subman_repos())
AssertionError: False is not true

----------------------------------------------------------------------
Ran 3 tests in 40.222s
```

**Comment by sbernhard** [comment]
thanks for you help. unfortunately I have no clue how to proceed as it don't really look like, they are related to my change. Do you have an hint / idea how to get more information whats broken?

**Pull request merged**